tests/test_optimizer.py

```diff
@@ -66,3 +69,116 @@ def test_grad_averager():
     # after no longer use_averaged_gradients
     assert not torch.allclose(model1.w.grad, ref_average)
     assert not torch.allclose(model2.w.grad, ref_average)
+
+
+@pytest.mark.forked
+@pytest.mark.parametrize("offload_optimizer, reuse_tensors", [(False, False), (True, False), (False, True)])
+def test_state_averager(offload_optimizer, reuse_tensors):
+    dht1 = hivemind.DHT(start=True)
+    dht2 = hivemind.DHT(initial_peers=dht1.get_visible_maddrs(), start=True)
+
+    torch.manual_seed(1337)
+    torch.use_deterministic_algorithms(True)
```

Comment:
nit: This affects the execution of all further tests. Instead, we can reset this flag to the previous value in the end of this test.

Reply:
Resolved by running test in a fork